Sources/Lifecycle/Lifecycle.swift

```diff
@@ -524,16 +485,30 @@ public extension ComponentLifecycle {
     ///    - label: label of the item, useful for debugging.
     ///    - start: `Handler` to perform the startup.
     ///    - shutdown: `Handler` to perform the shutdown.
-    func register(label: String, start: LifecycleHandler, shutdown: LifecycleHandler) {
-        self.register(Task(label: label, start: start, shutdown: shutdown))
+    public func register(label: String, start: LifecycleHandler, shutdown: LifecycleHandler) {
+        self.register(LifecycleTaskImpl(label: label, start: start, shutdown: shutdown))
     }
 
     /// Adds a `LifecycleTask` to a `LifecycleTasks` collection.
     ///
     /// - parameters:
     ///    - label: label of the item, useful for debugging.
     ///    - handler: `Handler` to perform the shutdown.
-    func registerShutdown(label: String, _ handler: LifecycleHandler) {
+    public func registerShutdown(label: String, _ handler: LifecycleHandler) {
         self.register(label: label, start: .none, shutdown: handler)
     }
 }
+
+internal struct LifecycleTaskImpl: LifecycleTask {
```

Comment:
Is it more common to name these `LifecycleTask: LifecycleTaskProtocol`?  `*Impl` is kind of Java-y.

Reply:
`LifecycleTask` is the public "currency type", while `LifecycleTaskImpl` is internal. While there are a few examples where Swift stdlib uses the `XxxProtocol` convention (e.g. `StringProtocol`) its actually not all that common to use it for public "currency type". That said I too not in love with `LifecycleTaskImpl` but since its internal I let it slide.

Reply:
It's not too bad if it's internal; true that making the actual public name good is more important 👍